src/env/set.c

```diff
@@ -0,0 +1,42 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*   tosh - 21 Shell                                      ::::::::            */
+/*                                                      :+:    :+:            */
+/*   By: aholster <aholster@student.codam.nl>          +:+                    */
+/*       ivan-tey <ivan-tey@student.codam.nl>         +#+                     */
+/*       nloomans <nloomans@student.codam.nl>        +#+                      */
+/*                                                 #+#    #+#                 */
+/*   License: GPLv3                                ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "env.h"
+
+int						env_set(t_env *const env,
+							char const *const key,
+							char const *const value)
+{
+	struct s_env_list	*new;
+
+	if (key == NULL || value == NULL)
+		return (-1);
```

Comment:
The `key` or `value` being `NULL` is a programming error and not something which should happen through normal execution. Either remove the `if` or replace it with an `assert`.